### PR TITLE
[GB-80] Add rate limit to API

### PR DESCRIPTION
**Pull request by nabhan-au**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -39,14 +43,34 @@ func main() {
...
 	app, err := firebase.NewApp(context.Background(), nil, opt)
 	if err != nil {
 		logrus.Fatalln("error initializing app: %v\n", err)
 	}
 
-	// Create the required dependentices
+	// Initialize redis
+	client := redis.NewClient(&redis.Options{
+		Addr:     appConfig.REDIS_CONNECTION_STRING,
+		Password: appConfig.REDIS_PASSWORD, // no password set
```
Comment (sirateek) [inline main.go]: Remove the unnecessary comment

File: main.go [hunk 2]
```diff
@@ -39,14 +43,34 @@ func main() {
...
+		DB:       0,                        // use default DB
```
Comment (sirateek) [inline main.go]: Could you also parameterize this plz?

**Review by sirateek — COMMENTED** [review]
Please see my comments.

**Comment by nabhan-au** [comment]
fixed

File: config/config.go [hunk 3]
```diff
@@ -14,6 +14,10 @@ type Config struct {
 	GIN_MODE                string `envconfig:"GIN_MODE" default:"release"`
 	MYSQL_CONNECTION_STRING string `envconfig:"MYSQL_CONNECTION_STRING"`
 	FIREBASE_CONFIG_FILE    string `envconfig:"FIREBASE_CONFIG_FILE" default:"firebase-credential.json"`
+	REDIS_CONNECTION_STRING string `envconfig:"REDIS_CONNECTION_STRING"`
+	REDIS_PASSWORD          string `envconfig:"REDIS_PASSWORD"`
+	RATE_LIMIT_STRING       string `envconfig:"RATE_LIMIT_STRING" default:"5-S"`
+	REDIS_DB                string `envconfig:"REDIS_DB" default:"0"`
```
Comment (sirateek) [inline config/config.go]: You may want to specify `REDIS_DB` as `int` type.

File: main.go [hunk 4]
```diff
@@ -39,14 +44,38 @@ func main() {
...
+	// Initialize the Firebase App
 	opt := option.WithCredentialsFile(appConfig.FIREBASE_CONFIG_FILE)
 	app, err := firebase.NewApp(context.Background(), nil, opt)
 	if err != nil {
 		logrus.Fatalln("error initializing app: %v\n", err)
 	}
 
-	// Create the required dependentices
+	// Initialize redis
+	redisDbStr, err := strconv.Atoi(appConfig.REDIS_DB)
```
Comment (sirateek) [inline main.go]: Since the type has been changed to int. So there is no need for this part.

**Review by sirateek — COMMENTED** [review]
Please see my comment

**Review by sirateek — APPROVED** [review]
LGTM

**Pull request merged**